cmd/generator/main_test.go

```diff
@@ -0,0 +1,86 @@
+package main
+
+import (
+	"testing"
+)
+
+type testCase struct {
+	name   string
+	regex  string
+	result string
+}
+
+func Test_generateCode(t *testing.T) {
+	var actual, expected, givenRegex string
+
+	tests := getTestData()
+	for _, tt := range tests {
+		expected = tt.result
+		givenRegex = tt.regex
+		t.Run(tt.name, func(t *testing.T) {
+			actual = generateCode(givenRegex)
+			if actual != expected {
+				t.Errorf("Expected:\n %s, \nGot:\n %s", expected, actual)
+			}
+		})
+	}
+}
+
+func getTestData() [3]testCase {
```

Comment:
What do you think about following signature?
```
func getSuccessfullTestCases() [3]testSuccessfullGenerating {
```

Reply:
It relates to the generator and it tests groups, doesn't it?

`func getGroupTestCases() [3]generatorTestCase {`

In the future, we will be able to organize functions for other sequences.

_Actually, we can avoid `array` here, because it would be easier to add a new test case with slice._

Reply:
Ok :) I changed it to ```func getGroupTestCases() []generatorTestCase {```